test/test_pub_sub.py.in

```diff
@@ -0,0 +1,256 @@
+      proc_info.assertWaitForShutdown(process=node_publisher_test, timeout=(@PUBSUB_TIMEOUT@ * 2))
+      proc_info.assertWaitForShutdown(process=node_relay_test, timeout=(@PUBSUB_TIMEOUT@ * 2))
+
+@launch_testing.post_shutdown_test()
+class PerformanceTestResults(unittest.TestCase):
+
+    def test_results_@TEST_NAME_PUB_SUB@(self, performance_log_prefix_cpumem_pub, performance_log_prefix_cpumem_sub, performance_log_prefix_pub, performance_log_prefix_sub, proc_info):
+        # self.addCleanup(_cleanUpLogs, performance_log_prefix_cpumem_pub)
+        # self.addCleanup(_cleanUpLogs, performance_log_prefix_cpumem_sub)
+        self.addCleanup(_cleanUpLogs, performance_log_prefix_pub)
+        self.addCleanup(_cleanUpLogs, performance_log_prefix_sub)
+
+        # launch_testing.asserts.assertExitCodes(
+        #     proc_info,
+        #     [launch_testing.asserts.EXIT_OK],
+        #     node_under_test,
+        # )
+
+        performance_logs = glob(performance_log_prefix_cpumem_pub + '*')
+        performance_logs_perf_test = glob(performance_log_prefix_pub + '_*')
+
+        if performance_logs and performance_logs_perf_test:
+            performance_report_csv = os.environ.get('PERFORMANCE_OVERHEAD_CSV')
+            if performance_report_csv:
+                _raw_to_csv(performance_logs[0], performance_logs_perf_test[0], performance_report_csv, "pub")
+            else:
+                print('No CSV report written - set PERFORMANCE_OVERHEAD_CSV to write a report',
+                      file=sys.stderr)
+
+            performance_report_png = os.environ.get('PERFORMANCE_OVERHEAD_PNG')
+            if performance_report_png:
+                _raw_to_png(performance_logs[0], performance_logs_perf_test[0], performance_report_png, "publisher", "@COMM_PUB@")
+            else:
+                print('No PNG report written - set PERFORMANCE_OVERHEAD_PNG to write a report',
+                      file=sys.stderr)
+        else:
+            print('No report written - no performance log was produced', file=sys.stderr)
+
+        performance_logs = glob(performance_log_prefix_cpumem_sub + '*')
+        performance_logs_perf_test = glob(performance_log_prefix_sub + '_*')
+
+        if performance_logs and performance_logs_perf_test:
+            performance_report_csv = os.environ.get('PERFORMANCE_OVERHEAD_CSV')
+            if performance_report_csv:
+                _raw_to_csv(performance_logs[0], performance_logs_perf_test[0], performance_report_csv, "sub")
+            else:
+                print('No CSV report written - set PERFORMANCE_OVERHEAD_CSV to write a report',
+                      file=sys.stderr)
+
+            performance_report_png = os.environ.get('PERFORMANCE_OVERHEAD_PNG')
+            if performance_report_png:
+                _raw_to_png(performance_logs[0], performance_logs_perf_test[0], performance_report_png, "subscriber", "@COMM_SUB@")
```

Comment:
```suggestion
                _raw_to_png(performance_logs[0], performance_logs_perf_test[0], performance_report_png, 'subscriber', '@COMM_SUB@')
```

Reply:
Actually, this whole file needs a lot of double-quotes converted to single-quotes.